documentation/tutorial.md

```diff
@@ -0,0 +1,74 @@
+# Git Town Tutorial
+
+Let's assume you are part of a development team with Bob and Carol.
+Your repository is hosted on GitHub, and you follow [GitHub Flow](https://guides.github.com/introduction/flow/index.html).
+
+
+## Starting a new feature
+
+You are in the middle of the sprint, and have just finished a feature.
+You take the next ticket from the backlog.
+Let's say it is called "resetting passwords".
+Since we are developing in feature branches, you now need to
+
+* update your master branch to the latest version on GitHub
+* cut a new feature branch from the master branch
+
+Running `git hack reset-passwords` achieves all this in a single command.
+This gives you the best possible start for building the new feature, 
+on top of the latest version of the code base.
+
+
+## Synchronizing the branch
+
+After coding for a while, you hear that Bob shipped a number of important bug fixes,
+and that Carol shipped some UI design updates.
+Both changes affect your work, so you want them in your feature branch before you continue.
+On a high level, you need to 
+
+* pull updates for the `master` branch (to get Bob and Carol's changes)
+* merge the master branch into your `reset-passwords` branch
+* push your updated feature branch to the repo, so that others who work on it also get these updates
+
+You will need to move between branches to do this, 
+which means you also need to stash away any currently open changes in your repo temporarily.
+All together this simple operation requires between 5 and 7 individual Git commands.
+That's a lot of typing. 
+And this should happen several times per day, for each of your feature branches!
+
+`git sync` runs this whole process in one command, and brings you back to exactly where you started.
+
+With Bob's bug fixes and the new UI from Carol available in your branch, 
+any more modification you make will fit right into their work.
+
+
+## Creating a pull request
+
+When you are done, you run `git pr`.
+It opens your browser with the GitHub page for creating a new pull request for your current branch.
+You fill out the title and description, tag the reviewers, and submit.
```

Comment:
This is the last section still in the old format:

-------------------
Once your feature is ready for review, its time to open a pull request on GitHub.
You fire up a browser, go to GitHub, navigate to your repository and finally end up on a new pull request form.

`git pr` lets you jump straight from the terminal to filling in the details of your pull request.
Your current branch is already selected, so all you need to do is fill out the title and description, tag the reviewers, and submit.